Library/Homebrew/command.rb

```diff
@@ -0,0 +1,157 @@
+module Homebrew
+  class Command
+    attr_reader :command_name, :valid_options, :description, :help_output,
+      :man_output
+
+    def initialize_variables
+      @valid_options = []
+      @root_options = []
+      @argv = ARGV.dup
+    end
+
+    def options(&block)
+      initialize_variables
+      @parent = nil
+      instance_eval(&block)
+      generate_help_and_manpage_output
+    end
+
+    def cmd_name(cmd)
+      @command_name = cmd
```

Comment:
@MikeMcQuaid, actually @woodruffw and I had a long discussion and did a of of debugging on this and came to the conclusion that `attr_accessor ` cannot be used for this because ruby does not recognize the `command_name` setter method (generated by `attr_accessor`) inside the child class i.e. `CommandsCommand`